Packages/vcs/Lib/vcsvtk/boxfillpipeline.py

```diff
@@ -0,0 +1,383 @@
+                self._contourLevels = \
+                      vcs.mkevenlevels(numpy.ma.log10(self._gm.level_1),
+                                       numpy.ma.log10(self._gm.level_2),
+                                       nlev=nlev)
+            else:
+                levslbls = vcs.mkscale(self._gm.level_1, self._gm.level_2)
+                self._contourLevels = vcs.mkevenlevels(self._gm.level_1,
+                                                       self._gm.level_2,
+                                                       nlev=nlev)
+            if len(self._contourLevels) > 25:
+                ## Too many colors/levels need to prettyfy this for legend
+                self._legend = vcs.mklabels(levslbls)
+                ## Make sure extremes are in
+                legd2 = vcs.mklabels([self._contourLevels[0],
+                                      self._contourLevels[-1]])
+                self._legend.update(legd2)
+            else:
+                self._legend = vcs.mklabels(self._contourLevels)
+            if self._gm.boxfill_type == "log10":
+                for k in self._legend.keys():
+                    self._legend[float(numpy.ma.log10(self._legend[k]))] = \
+                          self._legend[k]
+                    del(self._legend[k])
+
+        # Use consecutive colors:
+        self._contourColors = range(self._gm.color_1, self._gm.color_2 + 1)
+
+    def _updateContourLevelsAndColorsForCustomBoxfill(self):
+        """Set contour information for a custom boxfill."""
+        self._contourLevels = self._gm.levels
+
+        if numpy.allclose(self._contourLevels[0], [0., 1.e20]) or \
+              numpy.allclose(self._contourLevels, 1.e20):
+            levs2 = vcs.mkscale(self._scalarRange[0],
+                                self._scalarRange[1])
+            if len(levs2) == 1: # constant value ?
+                levs2 = [levs2[0], levs2[0] + .00001]
+            self._contourLevels = []
+            if vcs2vtk.isTrue(self._gm.ext_1):
+                ## user wants arrow at the end
+                levs2[0] = -1.e20
+            if vcs2vtk.isTrue(self._gm.ext_2):
+                ## user wants arrow at the end
+                levs2[-1] = 1.e20
+            for i in range(len(levs2) - 1):
+                self._contourLevels.append([levs2[i], levs2[i+1]])
+        else:
+            if not isinstance(self._gm.levels[0], (list, tuple)):
+                self._contourLevels = []
+                levs2 = self._gm.levels
+                if numpy.allclose(levs2[0], 1.e20):
+                    levs2[0] = 0
+                for i in range(len(levs2) - 1):
+                    self._contourLevels.append([levs2[i], levs2[i+1]])
+
+        # Contour colors:
+        self._contourColors = self._gm.fillareacolors
+        if self._contourColors is None:
+            # TODO BUG levs2 may not be defined here...
+            self._contourColors = vcs.getcolors(levs2, split=0)
```

Comment:
After simplifying the contour level logic, this bug became more obvious. I'll put in a test/bug report soon.